docs/configurations.md

```diff
@@ -23,7 +23,9 @@ summary: "Here we list all possible configurations and what they mean"
     <span style="color:grey">Should HoodieWriteClient autoCommit after insert and upsert. The client can choose to turn off auto-commit and commit on a "defined success condition"</span>
     - [withAssumeDatePartitioning](#withAssumeDatePartitioning) (assumeDatePartitioning = false) <br/>
         <span style="color:grey">Should HoodieWriteClient assume the data is partitioned by dates, i.e three levels from base path. This is a stop-gap to support tables created by versions < 0.3.1. Will be removed eventually </span>
-    
+    - [withMaxMemoryFractionPerPartitionMerge](#withMaxMemoryFractionPerPartitionMerge) (maxMemoryFractionPerPartitionMerge = 0.6) <br/>
+    <span style="color:grey">This fraction is multiplied with the spark.memory.fraction to get a final fraction of heap space to use during merge </span>
```

Comment:
We should also document how this interplays with `spark.memory.storageFraction` i.e this + that <= 1.0, otherwise things will OOM

Reply:
Actually, this is just defining what fraction of spark.memory.fraction is used for in-memory map in MergeHandle. So if the client uses sparkconfigs incorrectly, spark itself will cause issues before it hits the mergehandle code..

Reply:
I still think its worth documenting..
http://spark.apache.org/docs/2.1.1/tuning.html#memory-management-overview

 Usually, spark.memory.fraction - spark.memory.storageFraction is left for internal datastructures of spark. Heap - spark.memory.fraction is left for user data structures - in this case hoodie externalspillablemap.. 

I dont actually follow 
  - Why we multiply this fraction against spark fraction instead of total heap space. By definition, we cannot dip into the spark fraction
  - How can the spark app be stable is spark fraction is 0.6 and this also takes 0.6 * 0.6 of heap, which will bring total usage to 0.6 + 0.36 = 0.96 or 96% full heap, which will keep gcing back to back..

Reply:
The definitions in this link : https://spark.apache.org/docs/latest/configuration.html and the one you pasted (which is the description link in the above link) are a little misleading. I think there was an oversight on my part for choosing to multiple with spark.memory.fraction. Instead, I should multiply with 1 - spark.memory.fraction. It's more clear here : https://0x0fff.com/spark-memory-management/. 
I don't want to rely on heap space to calculate the spillablemap size since the heap is actually not the real user memory left by spark memory model. 
So my thought process is, executor.memory = heap. So as you increase executor.memory you increase the heap. We want our spillablemap memory size to grow accordingly, hence user.available.memory = executor.memory * (1 - spark.memory.fraction)
spillable.available.memory = user.available.memory*merge.memory.fraction.

@vinothchandar

Reply:
`spillable.available.memory = user.available.memory*merge.memory.fraction` sure, this is what I was getting at in the comments above. This should be okay.

Reply:
@n3nash lets reflect this discussion in the docs above?  Multiplying by spark fraction is definitely misleading

Reply:
done